app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -498,62 +503,118 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
             if (gattService == null) {
                 Log.w(TAG, "onServicesDiscovered: service " + xDripDataService + " not found");
                 listAvailableServices(mBluetoothGatt);
-                JoH.releaseWakeLock(wl);
-                return;
-            }
-
-            final BluetoothGattCharacteristic gattCharacteristic = gattService.getCharacteristic(xDripDataCharacteristic);
-            if (gattCharacteristic == null) {
-                Log.w(TAG, "onServicesDiscovered: characteristic " + xDripDataCharacteristic + " not found");
-                JoH.releaseWakeLock(wl);
-                return;
+                // try next
```

Comment:
it just searches the internal tables, once the services have been discovered, all services with their characteristics should be known to the ble stack.